make helm.generate doesn't work on macos

**Describe the solution you'd like**
`make helm.generate` is working on macos

**What is the added value?**
A developer can generate helm charts on his Apple laptop

**Observations (Constraints, Context, etc):**
```
❯ make helm.generate
sed: 1: "deploy/charts/external- ...": extra characters at the end of d command
sed: 1: "deploy/charts/external- ...": extra characters at the end of d command
sed: 1: "deploy/charts/external- ...": extra characters at the end of d command
sed: 1: "deploy/charts/external- ...": extra characters at the end of d command
make: *** [helm.generate] Error 1```

--- a/Makefile
+++ b/Makefile
@@ -206,8 +206,13 @@ helm.generate:
 		cat "$$i.bkp" >> "$$i" && \
 		echo "{{- end }}" >> "$$i" && \
 		rm "$$i.bkp" && \
-		sed -i 's/name: kubernetes/name: {{ include "external-secrets.fullname" . }}-webhook/g' "$$i" && \
-		sed -i 's/namespace: default/namespace: {{ .Release.Namespace | quote }}/g' "$$i" && \
+		if [[ "$$OSTYPE" == "darwin"* ]]; then \
+		  SEDPRG="gsed"; \
+		else \
+		  SEDPRG="sed"; \
+		fi; \
+		$$SEDPRG -i 's/name: kubernetes/name: {{ include "external-secrets.fullname" . }}-webhook/g' "$$i" && \
+		$$SEDPRG -i 's/namespace: default/namespace: {{ .Release.Namespace | quote }}/g' "$$i" && \
 		mv "$$i" "$${i%.yml}.yaml"; \
 	done
 	@$(OK) Finished generating helm chart files

